conda_env/specs/binstar.py

```diff
@@ -0,0 +1,111 @@
+import re
+from conda.resolve import normalized_version
+from .. import env
+from ..exceptions import EnvironmentFileDoesNotExist, EnvironmentFileNotDownloaded, CondaEnvException
+try:
+    from binstar_client import errors
+    from binstar_client.utils import get_binstar
+except ImportError:
+    raise CondaEnvException("Binstar not installed")
+
+ENVIRONMENT_TYPE = 'env'
+
+
+class BinstarSpec(object):
+    """
+    spec = BinstarSpec('darth/deathstar')
+    spec.can_handle() # => True / False
+    spec.environment # => YAML string
+    spec.msg # => Error messages
+    :raises: EnvironmentFileDoesNotExist, EnvironmentFileNotDownloaded
+    """
+
+    _environment = None
+    _username = None
+    _packagename = None
+    _package = None
+    _file_data = None
+    msg = None
+
+    def __init__(self, filename=None, name=None, **kwargs):
```

Comment:
I believe we need name here. https://github.com/conda/conda-env/blob/env-create-binstar/conda_env/specs/binstar.py#L53